lib/collection/src/collection.rs

```diff
@@ -258,10 +258,37 @@ impl Collection {
         &self,
         shard_id: ShardId,
         peer_id: PeerId,
-        active: ReplicaState,
+        state: ReplicaState,
     ) -> CollectionResult<()> {
         let shard_holder = self.shards_holder.read().await;
-        shard_holder.set_shard_replica_state(shard_id, peer_id, active)
+        let replica_set =
+            shard_holder
+                .get_shard(&shard_id)
+                .ok_or_else(|| CollectionError::NotFound {
+                    what: format!("Shard {shard_id}"),
+                })?;
+        replica_set.set_replica_state(&peer_id, state)?;
+
+        // Try to request shard transfer if replicas on the current peer are dead
+        if state == ReplicaState::Dead && self.this_peer_id == peer_id {
+            let transfer_from = replica_set
+                .peers()
```

Comment:
what about shuffling the peers before just to make sure we don't always ask the first one?

Reply:
This action here is not expected to be repeated. If transfer is declined, we are not retrying - that is the duty of external cluster manager.
Also HashSet is used there, it is sometimes chaotic